### DISCUSSION TITLE
TS Implementation for Sheet Indexes

### DISCUSSION THREAD

**Discussion by Ellord207** [iTwin/itwinjs-core#6993] [Ideas]
## Context:

My team is working on creating features to facilitate the production of 2d drawings and sheets for users using iTwin.js.  To do this, we need to be able to insert [Sheet Indexes](https://www.itwinjs.org/bis/domains/drawings-sheets/#sheet-index) and their relational elements into the iModel.  Since there are no Sheet Index TS implementations, we took the initiative and started creating them ourselves.
We have created a[ draft PR](https://github.com/iTwin/itwinjs-core/pull/6992) for these TS implementations and have a few questions about it.  I feel like larger discussion needs happen and the PR may not be the best place for that.  I will include my questions here and hope the right people will see it.

## Questions:

1.	Code
a.	How to add a BisCodeSpec?
b.	Is it needed for use case?
c.	What Scope Type is need?
2.	Should events such as [Element.onChildUpdate](https://www.itwinjs.org/reference/core-backend/elements/element/onchildupdatestatic/) manage the entryPriority of Sheet Index elements to ensure they are sequential?
3.	How do we handle user defined page labels?  There are scenarios where a user may want page number “%pagenumber%-a” or something like that.  Would DisplayLabel be appropriate?
4.	Do we need to write tests for these types where should they go? (Already started working on them porting them over from our development repo)
a.	Proposed location: core\backend\src\test\sheetIndex\SheetIndex.test.ts
b.	How to run tests for specific files, because running all backend tests takes long (Figured it out: `describe.only(....)`) .

**Comment by pmconne** [comment]
Most of these are questions for the BIS working group. @diegoalexdiaz may be able to help. He may want to schedule a conversation.

e.g., are sheet index priorities *supposed* to be sequential, and is it the job of core-backend to enforce that (somehow?) How do priorities work when one sheet index references another one via SheetIndexReference? SheetIndexFolders have their own priorities, and can contain other SheetIndexEntry's with their own priorities - how do priorities work for folders and down through the folder structure?

Re: tests - unless you intend to add multiple files, just put them in src/test/SheetIndex.test.ts.

**Comment by Ellord207** [comment]
I have marked the draft PR mentioned in this discussion as ready for review.  I intend to follow it up with another PR in the near future.
